### ISSUE TITLE
DIRSTACK isn't predefined in the POSIX shell

### ISSUE THREAD

**Issue by XSven** [koalaman/shellcheck#2750]
#### For bugs
- SC2178
- "online" 

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

DIRSTACK="$PWD"
export DIRSTACK
```

#### Here's what shellcheck currently says:

DIRSTACK="$PWD"
^-- [SC2178](https://www.shellcheck.net/wiki/SC2178) (warning): Variable was used as an array but is now assigned a string.

#### Here's what I wanted or expected to see:

No warning should be raised because DIRSTACK isn't predefined in the POSIX shell.
